### PR TITLE
fix(provider): record standalone calls and attribute fallback usage

### PR DESCRIPTION
**Pull request by RhoninSeiei**
Standalone SDK calls, plugin tool-loop agents and scheduled agents can return without appearing in provider statistics. Record these paths consistently, including errors and cancellations, and include them in Dashboard aggregates.

Attribute usage to the provider that performed each request, including successful tool rounds and schema-repair requests before a later fallback. Settle accumulated usage on provider switches without assigning earlier tokens to the final provider or counting them twice. Request-local ownership avoids duplicate provider and wrapper records.

Persist public token fields, retain usage attached to exceptions, and normalize unfinished execution times after fallback timing adjustments. With the current asynchronous runner initialization, early hook exits and image-preparation failures do not read an uninitialized runner. Cron records usage in its finalizer and treats an ERROR runner as a failed job.

Merged upstream master at 8b82d03 while preserving the original PR history. This PR remains independent of the OAuth provider PR. Validation: 283 related tests passed in an isolated Linux container, covering provider attribution, schema repair, errors, cancellation and unreset runners; Ruff lint and formatting checks passed.

## Summary by Sourcery

Record provider usage consistently across standalone calls and agent executions while preserving accurate attribution, failure handling, and Dashboard aggregates.

New Features:
- Record statistics for standalone provider SDK calls, tool-loop agents, and scheduled agents, including successful, failed, and cancelled executions.
- Attribute token usage and persisted statistics to the provider that handled each request, including fallback attempts and additional tool/schema-repair rounds.
- Include provider-level records in Dashboard token aggregates and success-rate calculations.

Bug Fixes:
- Prevent provider usage from being duplicated or attributed to the final fallback provider after provider switches.
- Preserve token usage from provider exceptions and normalize execution timing after fallback adjustments.
- Avoid accessing uninitialized runners during early pipeline exits and image-preparation failures.
- Treat cron runners ending in an ERROR state as failed jobs while still persisting their statistics.

Enhancements:
- Centralize provider and agent statistics recording with request-local ownership and public token-field normalization.

Tests:
- Add coverage for provider attribution across fallbacks, tool rounds, schema repair, errors, cancellation, cron execution, standalone calls, and runner initialization edge cases.

### CODE REVIEW

File: astrbot/core/provider/stats.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        usage = response.usage if response and response.usage else TokenUsage()
+        await db.insert_provider_stat(
+            umo=umo,
+            conversation_id=conversation_id,
+            provider_id=_provider_id(provider),
+            provider_model=provider.get_model(),
+            status=_response_status(response),
+            stats={
+                "token_usage": usage.__dict__.copy(),
+                "start_time": start_time,
```
[the remark this answers is not in the document]
Comment (RhoninSeiei) [reply astrbot/core/provider/stats.py]: 已在 `505191619` 处理。统计模块新增显式的 `_token_usage_dict()`，单次调用和后备 Provider 分段只向数据库传递 `input_other`、`input_cached`、`output`。同时增加动态额外属性测试，覆盖两种保存方式。这里保留为统计模块内部转换，避免扩大 `TokenUsage` 的公共接口。

File: tests/unit/test_star_context.py [hunk 2]
```diff
@@ -104,3 +145,155 @@ def test_add_llm_tools_handles_empty_tool_module_path():
...
+@pytest.mark.asyncio
+async def test_llm_generate_persists_one_provider_stat(temp_db):
+    provider = StatsProvider()
+    context = Context.__new__(Context)
+    context._db = temp_db
+    context.provider_manager = SimpleNamespace(
+        get_provider_by_id=AsyncMock(return_value=provider),
+    )
+
+    response = await context.llm_generate(
```
[the remark this answers is not in the document]
Comment (RhoninSeiei) [reply tests/unit/test_star_context.py:160]: 已在 `505191619` 补充两项测试：Provider 抛出 `RuntimeError` 时验证异常继续传播，并保存零 Token 的 `error` 记录；Provider 返回 `role=err` 时验证响应原样返回，并保存响应中的 Token 用量。相关用例均已通过。

**Comment by RhoninSeiei** [comment]
本次自动检查此前统一失败于 `anthropic==1.0.0` 将内部 HTTP 模块从 `httpx` 更名为 `httpx2`。本分支已同步上游 #9769 的同等兼容修复，并保留 Anthropic 0.x 支持；重新触发的 Unit Tests、Dashboard Build、格式检查、CodeQL 和多平台 Smoke Test 均已通过。#9769 合并后，这部分公共差异会从本 PR 中消失。
